rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+                )
+            })?;
+
+            if node_record.node_operator_id == new_operator_record.node_operator_principal_id {
+                println!(
+                    "{}do_replace_operator: Node {} already belongs to node operator {}",
+                    LOG_PREFIX, node_id, payload.new_operator_id
+                );
+                continue;
+            }
+
+            if node_record.node_operator_id != old_operator_record.node_operator_principal_id {
+                return Err(format!(
+                    "{}do_replace_operator: Node {} does not belong to node operator {}",
+                    LOG_PREFIX, node_id, payload.old_operator_id
+                ));
+            }
+
+            required_node_allowance += 1;
+            // Update the node record itself
+            let node_key = make_node_record_key(*node_id);
+            let updated_node_record = NodeRecord {
+                node_operator_id: new_operator_record.node_operator_principal_id.clone(),
+                ..node_record
+            };
+            mutations.push(update(node_key, updated_node_record.encode_to_vec()));
+        }
+
+        // Nothing should be done.
+        if mutations.is_empty() {
+            return Ok(());
+        }
+
+        if required_node_allowance > new_operator_record.node_allowance {
+            return Err(format!("{}do_replace_operator: Adding {} nodes would overflow node allowance for node operator {} who has {} remaining", LOG_PREFIX, required_node_allowance, payload.new_operator_id, new_operator_record.node_allowance));
+        }
+
+        // Update new node operator record to decrease node allowance
+        let new_node_operator_key = make_node_operator_record_key(payload.new_operator_id);
+        let updated_new_operator_record = NodeOperatorRecord {
+            node_allowance: new_operator_record.node_allowance - required_node_allowance,
+            ..new_operator_record.clone()
+        };
+        mutations.push(update(
+            new_node_operator_key,
+            updated_new_operator_record.encode_to_vec(),
+        ));
+
+        // Update old node operator record to increase node allowance
+        let old_node_operator_key = make_node_operator_record_key(payload.old_operator_id);
+        let updated_old_operator_record = NodeOperatorRecord {
+            node_allowance: old_operator_record.node_allowance + required_node_allowance,
+            ..old_operator_record.clone()
+        };
+        mutations.push(update(
+            old_node_operator_key,
+            updated_old_operator_record.encode_to_vec(),
+        ));
+
+        self.maybe_apply_mutation_internal(mutations);
+
+        println!(
+            "{}do_replace_operator: Finished executing payload: {:?}",
+            LOG_PREFIX, payload
+        );
+
+        Ok(())
+    }
+}
+
+fn find_node_operator_record_for_provider<'a>(
+    operators: &'a [NodeOperatorRecord],
+    operator_id: &'a PrincipalId,
```

Comment:
I don't think operator_id or provider need to be `&'a`... that's a surprising requirement

Reply:
You are right, its an overlook on my side. Will remove them, thanks!